app/forms/mappings/Formatters.scala

```diff
@@ -95,4 +95,30 @@ trait Formatters {
       override def unbind(key: String, value: A): Map[String, String] =
         baseFormatter.unbind(key, value.toString)
     }
+
+  private[mappings] def currencyFormatter(emptyErrorMessageKey: String, invalidErrorMessageKey: String): Formatter[BigDecimal] =
+    new Formatter[BigDecimal] {
+
+      val is2dp = """\d+|\d*\.\d{1,2}"""
+      val validNumeric = """[0-9.]*"""
+
+      private val baseFormatter = stringFormatter(emptyErrorMessageKey)
+
+      override def bind(key: String, data: Map[String, String]): Either[Seq[FormError], BigDecimal] =
+        baseFormatter
+          .bind(key, data)
+          .flatMap {
+            case s if s.isEmpty =>
+              Left(Seq(FormError(key, emptyErrorMessageKey)))
+            case s if !s.matches(validNumeric) || !s.matches(is2dp) =>
```

Comment:
it should be possible to merge this regExp expression into one, are we actually testing this logic ?

Reply:
This is something I just refactored away from the form that Daniel made. He took this from Submissions project do I'm not sure of the original intent of the ticket